game/src/test/java/com/hiddenswitch/spellsource/WitchDoctorTests.java

```diff
@@ -208,4 +213,253 @@ public void testWitchingTraveler() {
 			assertEquals(costOf(context, player, coin), 1);
 		});
 	}
-}
+
+	// Secretive Chanter: "Opener: Transform all Voodoo spells in your hand into Emerald Secrets."
+	@Test
+	public void testSecretiveChanter() {
+		runGym((context, player, opponent) -> {
+			Card v1 = receiveCard(context, player, "spell_divination");
+			Card v2 = receiveCard(context, player, "spell_frenzy");
+			Card v3 = receiveCard(context, player, "spell_hex_bolt");
+			Card v4 = receiveCard(context, player, "spell_spirit_bind");
+			Card v5 = receiveCard(context, player, "spell_spirit_bind");
+			Card nonv = receiveCard(context, player, "spell_fireball");
+			playMinionCard(context, player, "minion_secretive_chanter");
+			// Check whether all Voodoo spells have been transformed
+			assertTrue(player.getHand().get(0).getCardId().contains("secret_secret_of"));
+			assertTrue(player.getHand().get(1).getCardId().contains("secret_secret_of"));
+			assertTrue(player.getHand().get(2).getCardId().contains("secret_secret_of"));
+			assertTrue(player.getHand().get(3).getCardId().contains("secret_secret_of"));
+			assertTrue(player.getHand().get(4).getCardId().contains("secret_secret_of"));
+			assertEquals(player.getHand().get(5).getCardId(), "spell_fireball");
+		});
+	}
+
+	// High Shaman Mawliki: "Your spells have Toxic. Your healing is doubled. Your other minions have Elusive and Guard."
+	@Test
+	public void testHighShamanMawliki() {
+		runGym((context, player, opponent) -> {
+			Minion friend1 = playMinionCard(context, player, "minion_neutral_test");
+			Minion friend2 = playMinionCard(context, player, "minion_neutral_test_14");
+			playMinionCard(context, player, "minion_high_shaman_mawliki");
+			// Check for guard and elusive
+			assertTrue(friend1.hasAttribute(Attribute.TAUNT));
+			assertTrue(friend2.hasAttribute(Attribute.TAUNT));
+			assertTrue(friend1.hasAttribute(Attribute.UNTARGETABLE_BY_SPELLS));
```

Comment:
This sounds like an aura, so the card probably isn't correct at the moment. Otherwise you'd be checking for `Attribute.AURA_UNTARGETABLE_BY_SPELLS`, `Attribute.AURA_TAUNT`.

Reply:
test that when you remove the minion the attributes go away. but they should be aura attributes even if they do go away.